earn/src/pages/InfoPage.tsx

```diff
@@ -217,18 +217,24 @@ export default function InfoPage() {
       });
 
       // Get the time at which each pool was last updated via the oracle (using the update event and getLogs)
+      let updateLogs: ethers.providers.Log[] = [];
+      try {
+        updateLogs = await provider.getLogs({
+          address: ALOE_II_ORACLE_ADDRESS[chainId],
+          topics: [TOPIC0_UPDATE_ORACLE],
+          fromBlock: 0,
+          toBlock: 'latest',
+        });
+      } catch (e) {
+        console.error(e);
+      }
+      const filteredLogs = updateLogs.filter((log) => log.removed === false);
       const latestTimestamps = await Promise.all(
         poolAddresses.map(async (addr) => {
-          let updateLogs: ethers.providers.Log[] = [];
+          const associatedLogs = filteredLogs.filter((log) => log.topics[1] === `0x000000000000000000000000${addr}`);
```

Comment:
Assuming incoming logs are sorted chronologically (an assumption you're already making on line 237), we don't need to do a full filter. We can instead just find the _first matching element_, starting from the end of the array and going backwards.